ultralytics/data/split_dota.py

```diff
@@ -73,9 +72,9 @@ def load_yolo_dota(data_root, split="train"):
                     - val
     """
     assert split in ["train", "val"]
-    im_dir = os.path.join(data_root, f"images/{split}")
-    assert Path(im_dir).exists(), f"Can't find {im_dir}, please check your data root."
-    im_files = glob(os.path.join(data_root, f"images/{split}/*"))
+    im_dir = Path(data_root) / "images" / split
+    assert im_dir.exists(), f"Can't find {im_dir}, please check your data root."
+    im_files = glob(str(Path(data_root) / "images" / split / "*"))
```

Comment:
Could remove `glob` import and use `(data_root / f"images/{split}").glob("*")` returning a generator of `pathlib` objects here (or convert to a list but the `img2label_paths` function does that too).

Reply:
This would also mean that `img2label_paths` function would need to get updated, which looking at the `data.utils.py` file, has a far amount of `os` usage.

Reply:
@Burhan-Q yes in some strategic places os ops are left in place simply for speed, i.e. in the dataloaders it's much faster to return a thousand strings than a thousand Path objects. 

The main problem with Path().glob() is that it will likewise return Path() objects, maybe that would work too but this PR is just cleaning things up in place without modifying outputs.